crates/sui-framework/sources/tx_context.move

```diff
@@ -84,16 +84,18 @@ module sui::tx_context {
     }
 
     #[test_only]
-    /// Utility for creating 256 unique input hashes
+    /// Utility for creating 256 unique input hashes.
+    /// These hashes are guaranteed to be unique given a unique `hint: u64`
     fun dummy_tx_hash_with_hint(hint: u64): vector<u8> {
+        // split the 64 bits of hint into 2 bit increments
         let tx_hash = vector[];
-        let i = 1;
-        let hash_length = TX_HASH_LENGTH;
-        while (i <= hash_length) {
-            let value = i ^ hint % 256;
```

Comment:
@sblackshear I'm not sure the intent of this logic, but it cycles every 256 values of hint. These values from `dummy_tx_hash_with_hint` need to be unique, but happy to change this to any implementation that does that

Reply:
The intent of the logic was just that the old one crashed and I wanted it to not crash, so no issues with changing it :).

However, I think it would probably make sense to wait on @jonas-lj's https://github.com/MystenLabs/sui/pull/7554 + use the `hint` as a seed to his generator, just so we don't have to reproduce fancy generation in too many places.

Reply:
I suppose we could but it isn't really necessary as far as I can tell. Whats the thought behind using `sui::random` here?

Reply:
Only that this sort of logic is tricky (e.g., it's been changed 3 times due to various problems) and it's easier to centralize it in one place.

Reply:
I suppose my comment is that there is nothing here needing randomness. We just need some unique mapping of `u64 -> vector<u8>`. If anything we could just use `bcs::to_bytes` and then pad with zeros. I only didn't do that because the previous version tried to set each value. 

So thoughts on the implementation above or this BCS version:
```
let tx_hash = bcs::to_bytes(&hint);
while (vector::length(&tx_hash) < TX_HASH_LENGTH) vector::push_back(&mut tx_hash, 0);
tx_hash
```

Reply:
I have gone with the BCS approach